src/include/platform/PlatformManager.h

```diff
@@ -495,5 +500,20 @@ inline CHIP_ERROR PlatformManager::GetBootReasons(uint8_t & bootReasons)
     return static_cast<ImplClass *>(this)->_GetBootReasons(bootReasons);
 }
 
+inline CHIP_ERROR PlatformManager::GetActiveHardwareFaults(GeneralFaults<kMaxHardwareFaults> & hardwareFaults)
+{
+    return static_cast<ImplClass *>(this)->_GetActiveHardwareFaults(hardwareFaults);
+}
+
+inline CHIP_ERROR PlatformManager::GetActiveRadioFaults(GeneralFaults<kMaxRadioFaults> & radioFaults)
+{
+    return static_cast<ImplClass *>(this)->_GetActiveRadioFaults(radioFaults);
+}
+
+inline CHIP_ERROR PlatformManager::GetActiveNetworkFaults(GeneralFaults<kMaxNetworkFaults> & networkFaults)
+{
+    return static_cast<ImplClass *>(this)->_GetActiveNetworkFaults(networkFaults);
+}
+
```

Comment:
This should be from a new interface. We are adding too much to the PlatformManager.

Furthermore, this getter interface prevents supporting events about those faults.

Reply:
Adding a new platform dependent interface is quite expensive, including adding a new manager and generic implementation and all platforms need to implement the interface to override the default behaviors.

Reply:
Per sync offline, we will introduce DiagnosticDataProvider + DiagnosticsDelegate for all diagnostics clusters. Only handle these three attributes within this PR with new interface and will refactor all other diagnostics attributes to use this new interface in the followup PRs

Reply:
This PR has been merged, track the followup work at https://github.com/project-chip/connectedhomeip/issues/11922